src/rp2_common/hardware_rtc/rtc.c

```diff
@@ -122,6 +137,11 @@ static void rtc_irq_handler(void) {
 
     if (_alarm_repeats) {
         // If it is a repeatable alarm, re enable the alarm.
+        if(_alarm_repeats == CONTINUOUS_REPEAT_ON_SEC) {
+          datetime_t t;
+          rtc_get_datetime(&t);
+          rtc_hw->irq_setup_1 = RTC_IRQ_SETUP_1_SEC_ENA_BITS | ((((uint)t.sec + 1) % 60) << RTC_IRQ_SETUP_1_SEC_LSB);
```

Comment:
Is this change needed because otherwise the RTC alarm would just fire continuously?

Reply:
> Is this change needed because otherwise the RTC alarm would just fire continuously?

IRQ has been triggered on second x. This code is to renable it for second (x+1)%60. If not incremented, it would delay until it reaches second x again